x-pack/plugins/reporting/server/export_types/printable_pdf/lib/pdf/get_tempate.ts

```diff
@@ -0,0 +1,130 @@
+  title: string,
+  tableBorderWidth: number,
+  assetPath: string
+): Partial<TDocumentDefinitions> {
+  const pageMarginTop = 40;
+  const pageMarginBottom = 80;
+  const pageMarginWidth = 40;
+  const headingFontSize = 14;
+  const headingMarginTop = 10;
+  const headingMarginBottom = 5;
+  const headingHeight = headingFontSize * 1.5 + headingMarginTop + headingMarginBottom;
+  const subheadingFontSize = 12;
+  const subheadingMarginTop = 0;
+  const subheadingMarginBottom = 5;
+  const subheadingHeight = subheadingFontSize * 1.5 + subheadingMarginTop + subheadingMarginBottom;
+
+  return {
+    // define page size
+    pageOrientation: layout.getPdfPageOrientation(),
+    pageSize: layout.getPdfPageSize({
+      pageMarginTop,
+      pageMarginBottom,
+      pageMarginWidth,
+      tableBorderWidth,
+      headingHeight,
+      subheadingHeight,
+    }),
+    pageMargins: [pageMarginWidth, pageMarginTop, pageMarginWidth, pageMarginBottom],
+
+    header() {
+      return {
+        margin: [pageMarginWidth, pageMarginTop / 4, pageMarginWidth, 0],
+        text: title,
+        font: getFont(title),
+        style: {
+          color: '#aaa',
+        },
+        fontSize: 10,
+        alignment: 'center',
+      };
+    },
+
+    footer(currentPage: number, pageCount: number) {
+      const logoPath = path.resolve(assetPath, 'img', 'logo-grey.png'); // Default Elastic Logo
+      return {
+        margin: [pageMarginWidth, pageMarginBottom / 4, pageMarginWidth, 0],
+        layout: 'noBorder',
+        table: {
+          widths: [100, '*', 100],
+          body: [
+            [
+              {
+                fit: [100, 35],
+                image: logo || logoPath,
+              },
+              {
+                alignment: 'center',
+                text: i18n.translate('xpack.reporting.exportTypes.printablePdf.pagingDescription', {
+                  defaultMessage: 'Page {currentPage} of {pageCount}',
+                  values: { currentPage: currentPage.toString(), pageCount },
+                }),
+                style: {
+                  color: '#aaa',
+                },
+              },
+              '',
+            ],
+            [
+              logo
+                ? {
+                    text: i18n.translate(
+                      'xpack.reporting.exportTypes.printablePdf.logoDescription',
+                      {
+                        defaultMessage: 'Powered by Elastic',
+                      }
+                    ),
+                    fontSize: 10,
+                    style: {
+                      color: '#aaa',
+                    },
+                    margin: [0, 2, 0, 0],
+                  }
+                : '',
+              '',
+              '',
+            ],
+          ],
+        },
+      };
+    },
+
+    styles: {
+      heading: {
+        alignment: 'left',
+        fontSize: headingFontSize,
+        bold: true,
+        margin: [headingMarginTop, 0, headingMarginBottom, 0],
```

Comment:
Compare to https://github.com/elastic/kibana/blob/a74cea30c55e828ac5925f6daa083172c8d8f67d/x-pack/plugins/reporting/server/export_types/printable_pdf/lib/pdf/index.js#L268-L269

The previous code used invalid properties: `marginTop`, `marginBottom`

Reply:
Looks like the `margin(Left|Top|Right|Bottom)` is supported in the code, which means the previous code should be expected to work. So changing to `margin: [x, y, z, q]` should lead to the same result.

https://github.com/bpampuch/pdfmake/blob/b4b4906f7db27a78b33051cd90410aec4ec73478/tests/unit/DocMeasure.spec.js#L502